### ISSUE TITLE
SIP and WIP cause Deploy Problems task to fail

### ISSUE THREAD

**Issue by carlislemc** [picoCTF/picoCTF#470]
@royragsdale - after we merged in your pull request, I can't get SIP and WIP to work anymore. I get the below error. The command I ran was: 

J=4 M=8 WIP=192.168.2.52 SIP=192.168.2.53 vagrant up --provision shell && J=2 M=8 WIP=192.168.2.52 SIP=192.168.2.53 vagrant up --provision web

I still see a reference to 192.168.2.3

TASK [pico-shell : Deploy problems] ********************************************
fatal: [shell]: FAILED! => {
    "changed": true,
    "cmd": [
        "/picoCTF-env/bin/shell_manager",
        "deploy",
        "-n",
        "2",
        "all"
    ],
    "delta": "0:00:04.520829",
    "end": "2020-06-19 20:00:58.259945",
    "rc": 1,
    "start": "2020-06-19 20:00:53.739116"
}

STDERR:

20:00:54 hacksport.deploy INFO: default_user 'hacksports' does not exist. Creating the user now.
Traceback (most recent call last):
  File "/picoCTF-env/lib/python3.7/site-packages/urllib3/connectionpool.py", line 677, in urlopen
    chunked=chunked,
  File "/picoCTF-env/lib/python3.7/site-packages/urllib3/connectionpool.py", line 381, in _make_request
    self._validate_conn(conn)
  File "/picoCTF-env/lib/python3.7/site-packages/urllib3/connectionpool.py", line 976, in _validate_conn
    conn.connect()
  File "/picoCTF-env/lib/python3.7/site-packages/urllib3/connection.py", line 370, in connect
    ssl_context=context,
  File "/picoCTF-env/lib/python3.7/site-packages/urllib3/util/ssl_.py", line 390, in ssl_wrap_socket
    return context.wrap_socket(sock)
  File "/usr/lib/python3.7/ssl.py", line 423, in wrap_socket
    session=session
  File "/usr/lib/python3.7/ssl.py", line 870, in _create
    self.do_handshake()
  File "/usr/lib/python3.7/ssl.py", line 1139, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1076)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/picoCTF-env/lib/python3.7/site-packages/requests/adapters.py", line 449, in send
    timeout=timeout
  File "/picoCTF-env/lib/python3.7/site-packages/urllib3/connectionpool.py", line 725, in urlopen
    method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]
  File "/picoCTF-env/lib/python3.7/site-packages/urllib3/util/retry.py", line 439, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='192.168.2.3', port=2376): Max retries exceeded with url: /v1.35/_ping (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1076)')))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/picoCTF-env/bin/shell_manager", line 33, in <module>
    sys.exit(load_entry_point('ctf-shell-manager==1.2.1', 'console_scripts', 'shell_manager')())
  File "/picoCTF-env/lib/python3.7/site-packages/shell_manager/run.py", line 238, in main
    args.func(args)
  File "/picoCTF-env/lib/python3.7/site-packages/hacksport/deploy.py", line 1021, in deploy_problems
    restart_xinetd=False,
  File "/picoCTF-env/lib/python3.7/site-packages/hacksport/deploy.py", line 815, in deploy_problem
    deployment_directory=deployment_directory,
  File "/picoCTF-env/lib/python3.7/site-packages/hacksport/deploy.py", line 642, in generate_instance
    problem = Problem()
  File "/picoCTF-env/lib/python3.7/site-packages/hacksport/docker.py", line 44, in __init__
    self.client.ping()
  File "/picoCTF-env/lib/python3.7/site-packages/docker/client.py", line 187, in ping
    return self.api.ping(*args, **kwargs)
  File "/picoCTF-env/lib/python3.7/site-packages/docker/api/daemon.py", line 166, in ping
    return self._result(self._get(self._url('/_ping'))) == 'OK'
  File "/picoCTF-env/lib/python3.7/site-packages/docker/utils/decorators.py", line 46, in inner
    return f(self, *args, **kwargs)
  File "/picoCTF-env/lib/python3.7/site-packages/docker/api/client.py", line 230, in _get
    return self.get(url, **self._set_request_timeout(kwargs))
  File "/picoCTF-env/lib/python3.7/site-packages/requests/sessions.py", line 543, in get
    return self.request('GET', url, **kwargs)
  File "/picoCTF-env/lib/python3.7/site-packages/requests/sessions.py", line 530, in request
    resp = self.send(prep, **send_kwargs)
  File "/picoCTF-env/lib/python3.7/site-packages/requests/sessions.py", line 643, in send
    r = adapter.send(request, **kwargs)
  File "/picoCTF-env/lib/python3.7/site-packages/requests/adapters.py", line 514, in send
    raise SSLError(e, request=request)
requests.exceptions.SSLError: HTTPSConnectionPool(host='192.168.2.3', port=2376): Max retries exceeded with url: /v1.35/_ping (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1076)')))

**Comment by royragsdale** [comment]
@carlislemc , thank you for the report. Looks like the ENV values are incorrectly being passed into the vagrant machine where ansible is being run.

The `local_infra/inventory.yml` should pickup the env values: https://github.com/picoCTF/picoCTF/blob/master/infra_local/inventory.yml#L13

However it is clearly not as the configs that get generated have the defaults. For example

```
root@picoCTF-shell-dev:/opt# cat hacksports/local/local_config.json
{
    "hostname": "192.168.2.3",
    "web_server": "http://192.168.2.2",
    "rate_limit_bypass_key": "INSECURE_DEFAULT_CHANGE_ME"

        ,
    "docker_host": "tcp://192.168.2.3:2376",
    "docker_ca_cert": "/home/vagrant/.docker/ca.pem",
    "docker_client_cert": "/home/vagrant/.docker/cert.pem",
    "docker_client_key": "/home/vagrant/.docker/key.pem"

}
```

I believe that this can be fixed by explicitly setting the `ansible.extra_vars`. For example how the Vagrantfile used to pass things: https://github.com/picoCTF/picoCTF/blob/dd774b0f444cc7370de4787094907b83bde28462/Vagrantfile#L37

I'll test this fix and update/PR.

**Comment by dmartin** [comment]
Should be resolved via #471.
